src/components/components/CommonComponents.js

```diff
@@ -21,6 +22,14 @@ export class CommonComponents extends React.Component {
     entity: React.PropTypes.object
   };
 
+  componentDidMount () {
+    Events.on('selectedEntityComponentChanged', detail => {
+      if (DEFAULT_COMPONENTS.indexOf(detail.name) !== -1) {
+        this.forceUpdate();
```

Comment:
Yeah, if that works. `redux` might help later on if all the component data is in a single store.